### ISSUE TITLE
Some questions about the prediction

### ISSUE THREAD

**Issue by Zhou-openfoam** [maxjcohen/transformer#45]
First of all, thanks for your wonderful work. When I train my data with this model, the following problems arise:
![Loss](https://user-images.githubusercontent.com/48311584/104673845-81e70080-571d-11eb-9f06-6d1ab62baa5c.png)
![prediction](https://user-images.githubusercontent.com/48311584/104673925-a80ca080-571d-11eb-8775-c0fab28ec076.png)
The loss during training is accepted, but the difference between the predicted results and the true results is huge. The predicted results do not change over time.
In fact, no matter which model is used, the predicted results do not change over time.
For my datasets, the samples are 2000, the time sequence is 125, there are 11variables in X and one variable in Y.
Attached is my data.

[data.zip](https://github.com/maxjcohen/transformer/files/5818497/data.zip)
[dataset_22_125.zip](https://github.com/maxjcohen/transformer/files/5818498/dataset_22_125.zip)
[labels.zip](https://github.com/maxjcohen/transformer/files/5818501/labels.zip)

**Comment by maxjcohen** [comment]
Hi, your dataset doesn't seem to be normalized, do you train directly with the dataset you attached ? What are you preprocessing steps ?

**Comment by Zhou-openfoam** [comment]
Sorry for the late reply，I used the Min-Max Normalization in src/dataset.py to preprocess the data.
DATASET_PATH = 'datasets/data_22_125.npz'
ozeDataset = OzeDataset(DATASET_PATH)
dataset_train, dataset_val, dataset_test = random_split(ozeDataset, (3500, 1000, 500))

**Comment by Zhou-openfoam** [comment]
I think I've figured out it. If I want to get Yt, the input variables should be all Xt-1  and Yt -1

**Comment by maxjcohen** [comment]
Ok, so I'm assuming your rescaling your data before plotting, as your temperature is well above 1 here. I'm also assuming you're using your own dataset, as the temperature are quite high and not really smooth. So I'd encourage you to:

- Try your dataset with these more traditional networks that are usually able to train on pretty much anything, such as the LSTM. There is one implemented here on the repo, you can also use the one directly available in PyTorch.
- Try so good old fashion dataset debugging: plotting mean and std of your temperature at each hour, visualize input data, etc.
